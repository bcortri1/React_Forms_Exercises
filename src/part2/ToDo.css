.ToDo {
    display: block;
    justify-content: center;
    flex-wrap: wrap;
}

.remove-todo-btn {
    margin-right: 10px;
}