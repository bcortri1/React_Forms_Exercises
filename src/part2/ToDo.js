import React from 'react';
import "./ToDo.css";


const ToDo = ({id="test-id", text="", removeToDo}) => {
    const handleRemove = () => {
        removeToDo(id);
    }

    return <li className='ToDo' id={id}>
                <button className='remove-todo-btn' onClick={handleRemove}>X</button>{text}
            </li>;
};

export default ToDo;