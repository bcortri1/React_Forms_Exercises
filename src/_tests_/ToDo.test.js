import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ToDo from "../part2/ToDo";



// smoke test
it("renders without crashing", function () {
    render(<ToDo />);
});

// snapshot test
it("matches snapshot", function () {
    const { asFragment } = render(<ToDo />);
    expect(asFragment()).toMatchSnapshot();
});

