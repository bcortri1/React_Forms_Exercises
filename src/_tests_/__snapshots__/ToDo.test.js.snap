// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="ToDo"
    id="test-id"
  >
    <button
      class="remove-todo-btn"
    >
      X
    </button>
  </li>
</DocumentFragment>
`;
